apps/webservice/src/app/[workspaceSlug]/systems/[systemSlug]/deployments/[deploymentSlug]/releases/[versionId]/PolicyApprovalRow.tsx

```diff
@@ -42,28 +50,67 @@ export const PolicyApprovalRow: React.FC<PolicyApprovalRowProps> = ({
       router.refresh();
       utils.environment.policy.invalidate();
       utils.job.config.invalidate();
-      toast.success(`Approved release to ${environmentNames}`);
+      toast.success(`Approved release to ${environmentName}`);
     },
     onError: () => toast.error("Error approving release"),
   });
 
-  const environmentNames = environments.map((e) => e.name).join(", ");
-  const handleReject = () => rejectMutation.mutate({ releaseId, policyId });
-  const handleApprove = () => approveMutation.mutate(approval);
+  const handleReject = () =>
+    rejectMutation.mutate({
+      releaseId,
+      policyId,
+      userId: currentUserID!,
+    });
+  const handleApprove = () =>
+    approveMutation.mutate({
+      releaseId,
+      policyId,
+      userId: currentUserID!,
+    });
 
-  return (
-    <div className="flex items-center gap-2 rounded-md border border-blue-400/50 bg-blue-500/10 p-2 text-sm">
+  const renderStatusContent = () => {
+    if (status === "pending") {
+      return (
+        <>
+          <div className="flex items-center gap-2">
+            <Button
+              variant="destructive"
+              className="h-6 px-2"
+              onClick={handleReject}
+            >
+              Reject
+            </Button>
+            <Button className="h-6 px-2" onClick={handleApprove}>
+              Approve
+            </Button>
+          </div>
+        </>
+      );
+    }
+
+    if (status === "approved")
+      return (
+        <div className="ml-2 flex-grow">
+          <span className="font-medium">
+            <span className="text-green-300">Approved</span> by{" "}
+            {approval.user?.name}
+          </span>
+        </div>
+      );
```

Comment:
this is no different then defining react in react, move these to there own components